### PR TITLE
Add incremental sync via catalog

### PR DESCRIPTION
**Pull request by dscoleman**
To support incremental sync for streams the discovered catalog must be edited. This adds a catalog which extracts the Contact Salesforce API and performs INCREMENTAL sync against that endpoint.

Readme updated to explain procedure.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,15 +45,63 @@ The `api_type` is used to switch the behavior of the tap between using Salesforc
...
+> tap-salesforce --config config.json --catalog catalog.json [--state state.json]
 ```
 
+## Catalog, Extraction and Replication Types
+
+The included catalog file, `catalog.json`, is constructed to extract the Contact API and perform INCREMENTAL sync. 
+Refer to the [Singer spec](https://github.com/singer-io/getting-started/blob/master/docs/SYNC_MODE.md#replication-method) for 
+more information about replication types.
+
+By default, discovery mode will produce a catalog with forces FULL_TABLE sync for all streams.
```
Comment (timf-bonobos) [inline README.md]: nitpick
~~By default, discovery mode will produce a catalog with forces FULL_TABLE sync for all streams.~~
By default, discovery mode will produce a catalog that forces a FULL_TABLE sync for all streams.

File: README.md [hunk 2]
```diff
@@ -45,15 +45,63 @@ The `api_type` is used to switch the behavior of the tap between using Salesforc
...
+        ],
+        "table-key-properties": [
+          "Id"
+        ],
+        "selected": true
+      }
+    }
+    ```
+
+2. Add replication key to the catalog. Edit stream metadata in the catalog to identify the desired replication key.
```
Comment (timf-bonobos) [inline README.md]: By adding this `"replication-key": "SystemModstamp"` to the config tells the salesforce TAP that if this field changes then sync this?

**Review by timf-bonobos — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (dscoleman) [reply README.md]: You got it. The replication key in the catalog, if present, will cause the tap to compare the previous timestamp against the current. If current>previous then that record will be emitted and the state file will be updated with the new timestamp.

**Comment by dscoleman** [comment]
@timf-bonobos, thanks for having a look earlier. Anything else you spotted in your review?

**Review by timf-bonobos — APPROVED** [review]

**Pull request merged**